src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,264 @@
+    outlier_df = outlier_sample.reset_index()
+    outlier_df.columns = [ID_COL, outlier_col_label]
+    return outlier_df
+
+
+def read_all_outlier(filename_manta: str, filename_melt: str, filename_wham: str, outlier_type: str) -> pd.DataFrame:
+    """
+
+    Args:
+        filename_manta: Outliers determined in EvidenceQC for Manta.
+        filename_melt: Outliers determined in EvidenceQC for MELT.
+        filename_wham: Outliers determined in EvidenceQC for Wham.
+        outlier_type:
+
+    Returns:
+        The total number of times that a sample appears as an outlier
+        across all four algorithms for each sample.
+    """
+    # Manta:
+    col_name = get_col_name("manta", outlier_type)
+    outlier_manta_df = read_outlier(filename_manta, col_name)
+    dict_manta = dict(list(zip(outlier_manta_df[ID_COL], outlier_manta_df[col_name])))
+
+    # Melt:
+    col_name = get_col_name("melt", outlier_type)
+    outlier_melt_df = read_outlier(filename_melt, col_name)
+    dict_melt = dict(list(zip(outlier_melt_df[ID_COL], outlier_melt_df[col_name])))
+
+    # Wham:
+    col_name = get_col_name("wham", outlier_type)
+    outlier_wham_df = read_outlier(filename_wham, col_name, 11)
+    dict_wham = dict(list(zip(outlier_wham_df[ID_COL], outlier_wham_df[col_name])))
+
+    # merging all the dictionaries
+    outlier_dicts = [dict_manta, dict_melt, dict_wham]
+    merged_dicts = Counter()
+    for counted in outlier_dicts:
+        merged_dicts.update(counted)
+    all_outliers = dict(merged_dicts)
+    all_outliers_df = pd.DataFrame.from_dict(all_outliers, orient="index").reset_index()  # index_col=None
+    all_outliers_df.columns = [ID_COL, outlier_type + "_overall_outliers"]
+    return all_outliers_df
+
+
+def merge_evidence_qc_table(
+        filename_estimated_cn: str,
+        filename_mediancov: str,
+        filename_wgd: str,
+        filename_cnv_qvalues: str,
+        filename_high_manta: str,
+        filename_high_melt: str,
+        filename_high_wham: str,
+        filename_low_manta: str,
+        filename_low_melt: str,
+        filename_low_wham: str,
+        output_prefix: str) -> str:
+    """
+    Reads the provided TSV files (tab-delimited) and merges all the information in one table
+    serialized to the given output filename.
+    """
+    df_ploidy = read_ploidy(filename_estimated_cn)
+    df_bincov_median = read_bincov_median(filename_mediancov)
+    df_wgd_scores = read_wgd_scores(filename_wgd)
+    df_non_diploid = read_non_diploid(filename_cnv_qvalues)
+    df_manta_high_outlier = read_outlier(filename_high_manta, get_col_name("manta", "high"))
+    df_melt_high_outlier = read_outlier(filename_high_melt, get_col_name("melt", "high"))
+    df_wham_high_outlier = read_outlier(filename_high_wham, get_col_name("wham", "high"), 11)
+    df_total_high_outliers = read_all_outlier(filename_high_manta, filename_high_melt, filename_high_wham, "high")
```

Comment:
It is less efficient to read the file again from disk. Using the outputs of the previous function would decrease the number of times that you read from disk as well as the number of dataframe operations. You directly call and use read_outlier within read_all_outlier, so it would be very simple to make those dataframes inputs to read_all_outlier instead of calling the function again.

Reply:
We are in some sense using the output of other functions (`read_outlier`) inside the `read_all_outlier`. I don't think that making this change is necessary at this stage as I don't see much of a difference here.

Reply:
Let's touch base offline, I think it will be easier to explain!

Reply:
Sure! Meantime I thank you for your precise comment and conversation!
I think, for now, this is not going to cause any problems for us as I have checked the Pandas DataFrame memory usage for a different number of rows, columns, and data types. Even for the number of rows (in our case number of samples) in the order of 5e6 and the number of columns in the order of 40, we are not going to pass a couple of megabytes. 

So, I won't expect this to cause a problem for us, at the same time I definitely consider your comment for the future.